### ISSUE TITLE
[question] When does PPO2 verbose print out ep_reward_mean and ep_len_mean?

### ISSUE THREAD

**Issue by windweller** [hill-a/stable-baselines#927]
I'm using PPO2 to train a OpenAI gym environment LunarLander.

Here's my code, taken verbatim from the tutorial.

```python
import gym

from stable_baselines import PPO2
from stable_baselines.common.evaluation import evaluate_policy

env = gym.make('LunarLander-v2')

model = PPO2('MlpPolicy', env, learning_rate=1e-3, verbose=1)
model.learn(total_timesteps=int(2e5))
```

PPO2's code has two logging variables `ep_reward_mean` and `ep_len_mean` which are great to diagnose how well the agent is doing (in an environment where failure is common). If I just run this code normally, here's the logging I see:

```
--------------------------------------
| approxkl           | 0.00017229581 |
| clipfrac           | 0.0           |
| explained_variance | 0.313         |
| fps                | 637           |
| n_updates          | 260           |
| policy_entropy     | 0.95548433    |
| policy_loss        | -0.000601892  |
| serial_timesteps   | 33280         |
| time_elapsed       | 47.7          |
| total_timesteps    | 33280         |
| value_loss         | 88.99602      |
--------------------------------------
```

No sign of what I want...I tried to look into the condition `len(self.ep_info_buf) > 0 and len(self.ep_info_buf[0]) > 0` but this leads to `rollouts` and to `runner` (inherited from an abstract parent). Then I lost track.

How can I enable PPO2 to print out these two variables?

**Comment by araffin** [comment]
you can find the answer here: https://github.com/hill-a/stable-baselines/issues/24#issuecomment-422539183
it comes from the monitor wrapper.

**Comment by araffin** [comment]
also duplicate of #339 
this mentioned in the doc too: https://stable-baselines.readthedocs.io/en/master/guide/algos.html#rl-algorithms

**Comment by windweller** [comment]
Ahh I saw the issue "NaN" but didn't click on it...sorry about duplication.
I'm worried about using `Monitor` with `allow_early_reset` will cause some issue...so I'm currently ok with the tensorboard episode-reward setup.

Is there a tensorboard logging option for `episode length`?

**Comment by windweller** [comment]
Sorry about the replicated issue -- I did read the manual...but definitely not careful enough.

**Comment by araffin** [comment]
> Ahh I saw the issue "Mean Episode Reward and Length showing NaN" but didn't click on it...sorry about duplication.
> I'm worried about using `Monitor` with `allow_early_reset` will cause some issue...so I'm currently ok with the tensorboard episode-reward setup.
> 
> Is there a tensorboard logging option for `episode length`?

don't worry about the monitor, it is the recommended way to log training reward, episode reward is unfortunately buggy.

**Comment by ZZWang21** [comment]
I have a small question, I defined the "max_episode_time" as 1000 in gym. but in tensorboard, you can see the "ep_len_mean" is always 1000. is it normal? why? Thanks.
